src/Models/AZ3166Device.ts

```diff
@@ -315,118 +315,203 @@ export class AZ3166Device implements Device {
+          if (!deviceConnectionString) {
+            return false;
           }
-        ];
-      } else {
-        deviceConnectionStringSelection = [{
-          label: 'Input IoT Hub Device Connection String',
-          description: '',
-          detail: 'Input another...'
-        }];
-      }
+          if ((deviceConnectionString.indexOf('HostName') === -1) ||
+              (deviceConnectionString.indexOf('DeviceId') === -1) ||
+              (deviceConnectionString.indexOf('SharedAccessKey') === -1)) {
+            throw new Error(
+                'The format of the IoT Hub Device connection string is invalid. Please provide a valid Device connection string.');
+          }
+        }
 
-      const selection =
-          await vscode.window.showQuickPick(deviceConnectionStringSelection, {
-            ignoreFocusOut: true,
-            placeHolder: 'Choose IoT Hub Device Connection String'
-          });
+        if (!deviceConnectionString) {
+          return false;
+        }
 
-      if (!selection) {
-        return false;
-      }
+        console.log(deviceConnectionString);
+
+        // Set selected connection string to device
+        let res: boolean;
+        const plat = os.platform();
+        if (plat === 'win32') {
+          res = await this.flushDeviceConfig(deviceConnectionString, 0);
+        } else {
+          res =
+              await this.flushDeviceConfigUnix(deviceConnectionString, 0);
+        }
 
-      if (selection.detail === 'Input another...') {
+        if (res === false) {
+          return false;
+        } else {
+          vscode.window.showInformationMessage(
+              'Configure Device connection string successfully.');
+          return true;
+        }
+      } catch (error) {
+        throw error;
+      }
+    }else{
+      try {
         const option: vscode.InputBoxOptions = {
           value:
-              'HostName=<Host Name>;DeviceId=<Device Name>;SharedAccessKey=<Device Key>',
-          prompt: `Please input device connection string here.`,
-          ignoreFocusOut: true
+            '19e25a259d0c2be03a02d416c05c48ccd0cc7d1743458aae1cb488b074993eae',
+          prompt: `Please input UDS here.`,
+            ignoreFocusOut: true
         };
 
-        deviceConnectionString = await vscode.window.showInputBox(option);
-        if (!deviceConnectionString) {
+        let UDS = await vscode.window.showInputBox(option);
+        if (!UDS) {
           return false;
         }
-        if ((deviceConnectionString.indexOf('HostName') === -1) ||
-            (deviceConnectionString.indexOf('DeviceId') === -1) ||
-            (deviceConnectionString.indexOf('SharedAccessKey') === -1)) {
-          throw new Error(
-              'The format of the IoT Hub Device connection string is invalid. Please provide a valid Device connection string.');
+        let uds_test = /^(0[xX]){0,1}([A-Fa-f0-9]){64,64}$/;
+        if (uds_test.test(UDS) == false) {

```

Comment:
`showInputBox` option has validate function, you can put the validation in the option

Reply:
You should never use `==` in JS or TS, use `===`

![cuaymi4ueaavi2d](https://user-images.githubusercontent.com/1621293/42306419-78b6343a-8061-11e8-8731-360ce7114655.jpg)